liquidctl/driver/lianli_uni.py

```diff
@@ -0,0 +1,239 @@
+            raise ValueError(
+                f"channel must be between {_MIN_CHANNEL} and {_MAX_CHANNEL} (zero-based index)"
+            )
+
+        # Check if PWM is on for the channel
+        if self.pwm_channels[channel_index]:
+            _LOGGER.warning(
+                f"Cannot set fixed speed for Channel {channel_index + 1}: PWM is enabled. "
+                "Please disable PWM first to set a fixed speed."
+            )
+            return
+
+        duty = clamp(duty, 0, 100)
+        speed_byte = self._calculate_speed_byte(duty)
+        command = [224, channel_index + 32, 0, speed_byte]
+        _LOGGER.debug(
+            "Setting fixed speed for channel %d: duty %.1f%%, command %s",
+            channel_index,
+            duty,
+            command,
+        )
+        try:
+            self.device.write(command)
+        except Exception as e:
+            _LOGGER.error("Error writing to device: %s", e)
+        time.sleep(0.1)  # Delay to prevent race conditions
+
+        self.channel_speeds[channel_index] = duty
+
+        _LOGGER.info(f"Fan speed for Channel {channel_index + 1} set to {duty}%")
+
+    def _calculate_speed_byte(self, speed):
+        """Calculate the speed byte based on the device type and desired speed.
+
+        Parameters:
+            speed: int or float - The desired speed percentage (0-100)
+
+        Returns:
+            int - The calculated speed byte to send to the device
+        """
+        if self.device_type in ["SL", "AL"]:
+            if speed == 0:
+                return 40
+            else:
+                speed_byte = int((800 + (11 * speed)) / 19) & 0xFF
+        elif self.device_type == "SLI":
+            if speed == 0:
+                return 10
+            else:
+                speed_byte = int((250 + (17.5 * speed)) / 20) & 0xFF
+        elif self.device_type in ["SLV2", "ALV2"]:
+            if speed == 0:
+                return 7
+            else:
+                speed_byte = int((200 + (19 * speed)) / 21) & 0xFF
+        else:
+            raise NotSupportedByDevice(
+                f"Unsupported device type: {self.device_type}"
+            )
+        return speed_byte
+
+    def _extract_channel_index(self, channel):
+        """Extract the channel index from the channel name.
+
+        Parameters:
+            channel: str - The name of the channel (e.g., 'channel1')
+
+        Returns:
+            int - The zero-based index of the channel
+        """
+        match = re.search(r"(\d+)$", channel)
+        if match:
+            channel_number = int(match.group(1))
+            if 1 <= channel_number <= (_MAX_CHANNEL + 1):
+                return channel_number - 1
+        raise ValueError(
+            f"Invalid channel '{channel}'. Must be 'channel1' to 'channel{_MAX_CHANNEL + 1}'"
+        )
+
+    def disconnect(self, **kwargs):
+        """Disconnect from the device."""
+        _LOGGER.info("Disconnecting from device")
+        self.device.close()
```

Comment:
Unnecessary log entry, and therefore the entire override is also unneeded.

(If we were to log device disconnection, it would be better to do  it at the base class, for consistency. But I don't think we want to do it all).